api/v1alpha2/linodecluster_types.go

```diff
@@ -47,6 +47,11 @@ type LinodeClusterSpec struct {
 	// +optional
 	VPCRef *corev1.ObjectReference `json:"vpcRef,omitempty"`
 
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +optional
+	// FirewallRef is a reference to a firewall object. This makes the linode use the specified firewall.
+	FirewallRef *corev1.ObjectReference `json:"firewallRef,omitempty"`
+
```

Comment:
I think we want to rename this and firewallID so it is specific to just nodebalancers since other loadbalancers will not support this

Reply:
hmmm any thoughts on what we should rename it to? May be `LinodeFirewallRef`? 

Shouldn't we keep things consistent for the FirewallRef and FirewallID to what we are doing for Linode Machine?

Reply:
Since this firewall is used only by nodebalancers, we should rename it as `NodebalancerFirewallRef` or something similar. Someone might get confused seeing firewall ref in cluster and in linodemachine as they both can differ from one another.